### PR TITLE
Params: remove stage deps and --all option

### PR DESCRIPTION
**Pull request by dmpetrov**
Params:
1. No dependencies to stage files anymore
2. `--all` option for diff

Per iterative/dvc/pull/3631

### CODE REVIEW

File: content/docs/command-reference/params/diff.md [hunk 1]
```diff
@@ -7,7 +7,7 @@ commits in the <abbr>DVC repository</abbr>, or between a commit and the
 ## Synopsis
 
 ```usage
-usage: dvc params diff [-h] [-q | -v] [--show-json] [a_rev] [b_rev]
+usage: dvc params diff [-h] [-q | -v] [--all] [--show-json] [a_rev] [b_rev]
```
Comment (jorgeorpinel) [inline content/docs/command-reference/params/diff.md:10]: Got too long.
```suggestion
usage: dvc params diff [-h] [-q | -v] [--all] [--show-json]
                       [a_rev] [b_rev]
```

File: content/docs/command-reference/params/diff.md [hunk 2]
```diff
@@ -17,24 +17,24 @@ positional arguments:
 
 ## Description
 
-This command provides a quick way to compare parameter values from your previous
-experiments with the current one(s) in your <abbr>workspace</abbr>.
+This command provides a quick way to compare parameter values from your
+experiments in the repository history.
```
Comment (jorgeorpinel) [inline content/docs/command-reference/params/diff.md:21]:
```suggestion
This command provides a quick way to compare parameter values among experiments
in the repository history. Requires that Git is being used to version the
project params.
```

File: content/docs/command-reference/params/index.md [hunk 3]
```diff
@@ -45,9 +45,9 @@ parameters will be required for the stage invalidation (see `dvc status` and
 stage.
 
 Using parameter dependencies prevents situations where several
-[pipeline](/doc/command-reference/pipeline) stages depend on the same file, and
-any change in the file invalidates causes the reproduction all those stages
-unnecessarily.
+[pipeline](/doc/command-reference/pipeline) stages depend on the same
+configuration file, and any change in the file invalidates causes the
```
Comment (jorgeorpinel) [inline content/docs/command-reference/params/index.md:49]: I don't understand this change. What "configuration file"? I thought this sentence was about generic dependencies?

Maybe `...prevents situations where several pipeline stages share dependencies, and any change in those deps causes ..`

File: content/docs/command-reference/params/diff.md (hunk 1 above)
Comment (jorgeorpinel) [reply content/docs/command-reference/params/diff.md:10]: Imma just commit this one.

File: content/docs/command-reference/params/diff.md (hunk 2 above)
Comment (jorgeorpinel) [reply content/docs/command-reference/params/diff.md:22]: > p.s. I'm applying similar changes to `metrics diff` in https://github.com/iterative/dvc.org/pull/1131/files#diff-8c12fa5ffd3e179ae1be9ed9eb448622

**Comment by shcheklein** [comment]
@dmpetrov @efiop it looks we haven't updated completion scripts for `params`

@efiop should we make a checkbox after all?

**Comment by dmpetrov** [comment]
> @dmpetrov @efiop it looks we haven't updated completion scripts for `params`

@shcheklein could you please clarify?

**Comment by shcheklein** [comment]
@dmpetrov disregard it here, please, I'll create a ticket on the dvc repo. it's not related to docs.

**Pull request merged**